pandas/core/internals/managers.py

```diff
@@ -1947,9 +1947,17 @@ def _blklocs(self):
         """compat with BlockManager"""
         return None
 
-    def getitem_mgr(self, indexer: slice | npt.NDArray[np.bool_]) -> SingleBlockManager:
+    def getitem_mgr(self, indexer: slice | np.ndarray) -> SingleBlockManager:
```

Comment:
The type hint is incorrect, we can also get here with an int indexer

Reply:
If this type hint is incorrect, then also the one in `Series._get_values` (since that is the only place where this is being used)

Reply:
I only see `getitem_mgr` used in `Series._get_values`, and this in itself is only used in `Series._slice` and `Series.__getitem__` after a `com.is_bool_indexer(key)` check. 
So then `Series._slice` should actually be used for more than a slice and its type annotation is also wrong?

Reply:
Yes, it seems that `_convert_slice_indexer` in 

https://github.com/pandas-dev/pandas/blob/99859e4e3d2d367cb58e9bfbd61731abeaaac790/pandas/core/series.py#L960-L964

is not guaranteed to return a slice object (only in case of DatetimeIndex, I think, through `slice_indexer`)

Reply:
Will look more closely over the next days

Reply:
Yep you are correct. We can get there via both paths, so np.ndarray is probably correct.

Update _slice as well